From fae03322bb532fbe7e2cedd9f3c4e83fd229a78b Mon Sep 17 00:00:00 2001
From: "Wu, Hao" <hao.wu@intel.com>
Date: Sat, 16 May 2015 17:53:40 +0800
Subject: [PATCH] pci quirks: disable pme poll for cherrytrail xhci controller

PME wakeup works well for CHT XHCI controller, so no need to
do 1s interval PME Polling, which may impact system low power
state.

Change-Id: I5675d0d6a8e3b1cadc8a3c5470c936af10507398
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-36520
Signed-off-by: Wu, Hao <hao.wu@intel.com>
---
 drivers/pci/quirks.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/pci/quirks.c b/drivers/pci/quirks.c
index b89ba6b06d13..fcabd856587e 100644
--- a/drivers/pci/quirks.c
+++ b/drivers/pci/quirks.c
@@ -2946,6 +2946,12 @@ static void quirk_pcie_enable_rtpm_ignore_children(struct pci_dev *dev)
 DECLARE_PCI_FIXUP_ENABLE(PCI_VENDOR_ID_INTEL,
 	PCI_DEVICE_ID_INTEL_CHV_PCIe_1, quirk_pcie_enable_rtpm_ignore_children);
 
+static void quirk_disable_pme_poll(struct pci_dev *dev)
+{
+	pci_disable_dev_pme_poll(dev, NULL);
+}
+DECLARE_PCI_FIXUP_ENABLE(PCI_VENDOR_ID_INTEL, 0x22b5, quirk_disable_pme_poll);
+
 static ktime_t fixup_debug_start(struct pci_dev *dev,
 				 void (*fn)(struct pci_dev *dev))
 {
-- 
1.9.1

